Server/App/model/discover.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+from model.deck import Deck
+from utils.database import DatabaseConnector
+from utils.serializer import serializer
+
+
+class Discover(object):
+    def __init__(self, user_id, connector=None):
+        self.user_id = user_id
+        self.connector = connector or DatabaseConnector()
+
+    def fetch(self, category, page=1):
+        deck_uuids = self.connector.call_procedure('FETCH_DECK_UUIDS', category, page)
+        results = []
+        for deck_uuid in deck_uuids:
+            results.append(Deck(deck_uuid['uuid'], connector=self.connector).get_metadata(self.user_id))
```

Comment:
I totally hear what you're saying but my theory was that with tags, we had to do an extra request for each individual deck to retrieve the tags anyway. So I figured its still the same number of queries. Also, it allows us to only have to maintain the DECK_FETCH procedure, so now we don't have a bunch of code duplication on the SQL side, which I'm happy with, because maintaining stored procedures sucks.